test/Database/LSMTree/Model/Normal.hs

```diff
@@ -225,3 +229,36 @@ duplicate ::
      Table k v blob
   -> Table k v blob
 duplicate = id
+
+{-------------------------------------------------------------------------------
+  Cursor
+-------------------------------------------------------------------------------}
+
+type Cursor :: Type -> Type -> Type -> Type
+data Cursor k v blob = Cursor
+    { _cursorValues :: [(RawBytes, (RawBytes, Maybe (BlobRef blob)))]
```

Comment:
It's kind of clear from the usage of `Map.toList` already, but might be worth adding a short note that these entries are already resolved, or more generally that the first element of the tuple is the key and entries are sorted+unique by key.